pkg/controller/ingress-shim/sync.go

```diff
@@ -183,13 +185,7 @@ func (c *controller) buildCertificates(ctx context.Context, ing *extv1beta1.Ingr
 
 			updateCrt := existingCrt.DeepCopy()
 
-			updateCrt.Spec.DNSNames = tls.Hosts
-			updateCrt.Spec.SecretName = tls.SecretName
-			updateCrt.Spec.IssuerRef.Name = issuerName
-			updateCrt.Spec.IssuerRef.Kind = issuerKind
-			updateCrt.Spec.IssuerRef.Group = issuerGroup
-			updateCrt.Spec.CommonName = ""
-			updateCrt.Labels = ing.Labels
+			updateCrt.Spec = crt.Spec
```

Comment:
I almost lgtm'ed.. But spootted another thing we mised in the hurry of the call.. `updateCert.Labels` is not updated by `crt.Spec`
Also looks like why the tests are failing

Reply:
Good spot!